### ISSUE TITLE
Allow specifying an alternative storage backend for image renditions

### ISSUE THREAD

**Issue by gasman** [torchbox/wagtail#3183]
The `file` field of AbstractRendition doesn't specify a `storage` parameter, so it will be fixed as Django's DEFAULT_FILE_STORAGE setting. We should make this a configuration setting (suggested: `WAGTAILIMAGES_RENDITION_STORAGE`, defaulting to DEFAULT_FILE_STORAGE).

This would allow renditions to be stored in a distinct filesystem location from other uploaded files, outside of MEDIA_ROOT - for example, a `/public` folder configured at the webserver level. (It would probably also give us a more sensible setup for hosting images on services such as S3 - that way the original images would still exist locally, so we wouldn't need a download-then-upload round trip when generating renditions.)

**Comment by lb-** [comment]
Marking this as a good first issue.

* wagtail/images/models.py contains the `AbstractRendition`
* create & document a new images setting `WAGTAILIMAGES_RENDITION_STORAGE` in docs/reference/settings.rst
* Needs unit tests

**Comment by eevelweezel** [comment]
Submitted https://github.com/wagtail/wagtail/pull/8155

**Comment by lb-** [comment]
Here is the documentation on the `storage` param for `FileField`
https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.FileField.storage

Here is more info about Django file storage
https://docs.djangoproject.com/en/4.0/topics/files/#file-storage

Rough implementation

```
from django.core.files.storage import default_storage

# ...

class AbstractImage(ImageFileMixin, CollectionMember, index.Indexed, models.Model):
    title = models.CharField(max_length=255, verbose_name=_("title"))
    file = models.ImageField(
        verbose_name=_("file"),
        upload_to=get_upload_to,
        storage=get_storage, # added
        width_field="width",
        height_field="height",
    )

    # ... other methods

    get_storage(self):
        return getattr(settings, "WAGTAILIMAGES_RENDITION_STORAGE", default_storage)

```

See Django FileField implementation for more details if needed https://github.com/django/django/blob/stable/4.0.x/django/db/models/fields/files.py

**Comment by eevelweezel** [comment]
I agree, this setting makes more sense as a custom storage object vs. a path.  PR is updated.
